include/TradingEngine/order_tree.h

```diff
@@ -0,0 +1,69 @@
+/* File: order_tree.h
+
+Contains declaration for composition class responsible for managing objects of LimitOrder class 
+through a balanced binary search tree (red-black tree). The Tree will be sorted based on Price only, 
+other factors like event time are not considered. The class also contains data structures for efficient 
+access to LimitOrder and Order objects.
+*/
+
+#ifndef TRADING_ENGINE_SOURCE_HEADERS_ORDER_TREE_H_
+#define TRADING_ENGINE_SOURCE_HEADERS_ORDER_TREE_H_
+
+#pragma once
+#include <map>
+#include <unordered_map>
+#include <memory>
+
+#include "transactions.h"
+#include "order.h"
+#include "limit_order.h"
+
+
+class OrderTree {
+private:
+  std::map<float, int> order_tree_; /* Map (red-black tree) with LimitOrder price as key and LimitOrder length as value */
```

Comment:
1. Why do you need a map instead of unordered_map?
2. This map will not work correctly, because of the floating-point precision  problem in C++, there are work arounds though. You can use something like this to convert the float to a string to be able to store safely:
```
std::string floatToString(float value, int precision) {
    std::ostringstream ss;
    ss << std::fixed << std::setprecision(precision) << value;
    return ss.str();
}
```

Reply:
This is fine - no need to change. Read more into this. Maybe round down orders to some precision point when they are submitted.